### PR TITLE
mons: improve mon restore quorum

### PR DESCRIPTION
**Pull request by subhamkrai**
validating user input `yes-really-restore` and also fixing a couple of typos.

fixes: #97

### CODE REVIEW

File: pkg/mons/restore_quorum.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func restoreQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace
 	logging.Info("The cluster fsid is %s\n", cephFsid)
 
 	var answer, output string
-	logging.Warning("Are you sure you want to restore the quorum to mon %s? If so, enter 'yes-really-restore\n", goodMon)
+	logging.Warning("Are you sure you want to restore the quorum to mon %s? If so, enter 'yes-really-restore`\n", goodMon)
```
Comment (travisn) [inline pkg/mons/restore_quorum.go:87]:
```suggestion
	logging.Warning("Are you sure you want to restore the quorum to mon %s? If so, enter 'yes-really-restore'\n", goodMon)
```

File: pkg/mons/restore_quorum.go [hunk 2]
```diff
@@ -279,13 +279,13 @@ func getMonDetails(goodMon string, monEndpoints []string) ([]string, string, str
 func promptToContinueOrCancel(answer string) (string, error) {
 	var ROOK_PLUGIN_SKIP_PROMPTS string
 	_, ok := os.LookupEnv(ROOK_PLUGIN_SKIP_PROMPTS)
-	if ok {
+	if !ok {
 		if answer == "yes-really-restore" {
 			return "proceeding", nil
 		} else if answer == "" {
 			return "continuing", nil
```
Comment (travisn) [inline pkg/mons/restore_quorum.go:286]: If the answer is blank we don't want to continue. They have to type the string to continue.

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/mons/restore_quorum.go (hunk 2 above)
Comment (subhamkrai) [reply pkg/mons/restore_quorum.go]: done

**Review by travisn — COMMENTED** [review]
Now the CI is failing with this change

File: pkg/mons/restore_quorum.go [hunk 3]
```diff
@@ -277,15 +277,12 @@ func getMonDetails(goodMon string, monEndpoints []string) ([]string, string, str
 }
 
 func promptToContinueOrCancel(answer string) (string, error) {
-	var ROOK_PLUGIN_SKIP_PROMPTS string
-	_, ok := os.LookupEnv(ROOK_PLUGIN_SKIP_PROMPTS)
-	if ok {
+	_, ok := os.LookupEnv("ROOK_PLUGIN_SKIP_PROMPTS")
```
Comment (travisn) [inline pkg/mons/restore_quorum.go:280]: Instead of just seeing if the env var exists, let's check if it also equals `true`.

Comment (subhamkrai) [reply pkg/mons/restore_quorum.go]: done

**Review by travisn — APPROVED** [review]

File: pkg/mons/restore_quorum.go [hunk 4]
```diff
@@ -277,17 +277,15 @@ func getMonDetails(goodMon string, monEndpoints []string) ([]string, string, str
...
-	var ROOK_PLUGIN_SKIP_PROMPTS string
-	_, ok := os.LookupEnv(ROOK_PLUGIN_SKIP_PROMPTS)
-	if ok {
-		if answer == "yes-really-restore" {
-			return "proceeding", nil
-		} else if answer == "" {
-			return "continuing", nil
-		} else {
-			return "", fmt.Errorf("canncelled")
+	if skip, ok := os.LookupEnv("ROOK_PLUGIN_SKIP_PROMPTS"); ok {
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply pkg/mons/restore_quorum.go]: it's better

**Pull request merged**